sklearn/utils/tests/test_validation.py

```diff
@@ -992,12 +993,21 @@ class WrongDummyMemory:
     pass
 
 
-@pytest.mark.filterwarnings("ignore:The 'cachedir' attribute")
 def test_check_memory():
+    joblib_before_0_12 = parse_version(joblib.__version__) < parse_version("0.12")
+
     memory = check_memory("cache_directory")
-    assert memory.cachedir == os.path.join("cache_directory", "joblib")
+    if joblib_before_0_12:
+        assert memory.cachedir == os.path.join("cache_directory", "joblib")
+    else:
+        assert memory.location == "cache_directory"
+
     memory = check_memory(None)
-    assert memory.cachedir is None
+    if joblib_before_0_12:
```

Comment:
Might be a good time to upgrade minimal required joblib, particularly if we are not testing it? Currently, it's 0.11 released in 2017 which is a long time ago.

Reply:
We only test the minimum version of joblib in the ubuntu_altas job and coverage is turned off there.

I'm okay with bumping up the version. Do we go straight to joblib `1.1` or go to `0.12.5`?

Reply:
Not sure. There were some known issues in the first few releases when joblib included loki around 0.12, maybe we can just go with 1.0 directly (late 2020)?

@ogrisel any opinion on this?

Reply:
I think 1.0 is fine.

Reply:
Updated PR to bump joblib dependency to 1.0.0, which also lead to deleting a bunch of code.